### PR TITLE
fix: correct URL for entry points

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

There were two issues:
- the test case was using `for..in` instead of `for..of`, because of that, the test was running but the for loop wasn't executed. Fixing that showed the issue and it was failing;
- fixed the path of `entryPoints`, which now uses the correct `build.server` URL
- we forgot to document the introduction of `entryPoints` in a changeset; we can't create a new changeset now because the feature is already released. I changed `CHANGELOG.md`. Once accepted, I will change the text of https://github.com/withastro/astro/releases/tag/astro%402.7.0

## Testing

Fixed the current test case

## Docs

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

**Comment by ematipico** [comment]
cc @alexanderniebuhr

File: packages/astro/src/@types/astro.ts [hunk 1]
```diff
@@ -111,21 +111,6 @@ export interface CLIFlags {
...
-	/**
-	 * @deprecated Use config.build.server instead.
-	 */
-	server: URL;
-	/**
-	 * @deprecated Use config.build.serverEntry instead.
-	 */
-	serverEntry: string;
-}
-
```
Comment (bluwy) [inline packages/astro/src/@types/astro.ts:128]: Love this refactor!

File: packages/astro/CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,24 @@
     },
   });
   ```
+- [#7220](https://github.com/withastro/astro/pull/7220) [`459b5bd05`](https://github.com/withastro/astro/commit/459b5bd05f562238f7250520efe3cf0fa156bb45) Thanks [@ematipico](https://github.com/ematipico)! - Shipped a new SSR build configuration mode: `split`.
+
+  The Astro hook `astro:build:ssr` now receives a new option in their payload, called `entryPoints`.
```
Comment (bluwy) [inline packages/astro/CHANGELOG.md:40]: Tiny nit following how changesets usually format these:
```suggestion

- [#7220](https://github.com/withastro/astro/pull/7220) [`459b5bd05`](https://github.com/withastro/astro/commit/459b5bd05f562238f7250520efe3cf0fa156bb45) Thanks [@ematipico](https://github.com/ematipico)! - The Astro hook `astro:build:ssr` now receives a new option in their payload, called `entryPoints`.
```

1. Prepend a new line
2. Made `The Astro hook ...` the first changesets message instead.

**Review by bluwy — APPROVED** [review]
One small nit, but otherwise looks great!

NOTE: The changelog edit sounds fine to me. IIRC you can do that with changesets. Editing the release later should also be fine too.

File: packages/astro/src/@types/astro.ts (hunk 1 above)
Comment (ematipico) [reply packages/astro/src/@types/astro.ts:128]: Yeah it was long overdue

**Pull request merged**